pkg/skaffold/cluster/minikube.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/homedir"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/context"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+var GetClient = getClient
+
+// To override during tests
+var (
+	minikubeBinaryFunc      = minikubeBinary
+	getRestClientConfigFunc = context.GetRestClientConfig
+	getClusterInfo          = context.GetClusterInfo
+)
+
+const (
+	VirtualBox = "virtualbox"
+	HyperKit   = "hyperkit"
+)
+
+type Client interface {
+	// IsMinikube returns true if the given kubeContext maps to a minikube cluster
+	IsMinikube(kubeContext string) bool
+	// MinikubeExec returns the Cmd struct to execute minikube with given arguments
+	MinikubeExec(arg ...string) (*exec.Cmd, error)
+}
+
+type clientImpl struct{}
+
+func getClient() Client {
+	return clientImpl{}
+}
+
+func (clientImpl) IsMinikube(kubeContext string) bool {
+	// short circuit if context is 'minikube'
+	if kubeContext == constants.DefaultMinikubeContext {
+		return true
+	}
+	if _, err := minikubeBinaryFunc(); err != nil {
+		logrus.Tracef("Minikube cluster not detected: %v", err)
+		return false
+	}
```

Comment:
Should this check be performed first?  If we can't find the minikube binary, then it doesn't really matter if `context == 'minikube'`, since we can't run `minikube docker-env`?

Reply:
Fixed in https://github.com/GoogleContainerTools/skaffold/pull/4742